src/v0/destinations/ga4/utils.js

```diff
@@ -432,11 +433,39 @@ const prepareUserProperties = (message, piiPropertiesToIgnore = []) => {
   return validatedUserProperties;
 };
 
+/**
+ * Returns user consents
+ * Ref : https://developers.google.com/analytics/devguides/collection/protocol/ga4/reference?client_type=gtag#payload_consent
+ * @param {*} message
+ * @returns
+ */
+const prepareUserConsents = (message) => {
+  const consents = {};
+  const GOOGLE_ALLOWED_CONSENT_STATUS = ['GRANTED', 'DENIED'];
+  const integrationObj = getIntegrationsObj(message, 'ga4') || {};
+  const eventLevelConsentsData = integrationObj?.consents || {};
+
```

Comment:
@mihir-4116 update your base to `feat.consentMode.GAOC` so that you get latest googleUtil implementation.
We planned to tackle all consent changes related to google destinations in the above util to avoid code deduplication wherever same consent logic is needed.

1. Define configMap, ref: https://github.com/rudderlabs/rudder-transformer/pull/3121/files#diff-8fb010420f7633935159c0f5c015492a7941d65f1a7368e4c9b2dfb08f1d0b2cR45
const consentConfigMap = {
  analyticsPersonalizationConsent: 'adPersonalization',
  analyticsUserDataConsent: 'adUserData',
};

2. Merge this logic in `populateConsentForGoogleDestinations` function - https://github.com/rudderlabs/rudder-transformer/blob/51005d8a7c4eec49bdb487b807ac1e7f3d0ea6e0/src/v0/util/googleUtils/index.js#L14

Reply:
cc: @saikumarrs